Xamarin.Essentials/TextToSpeech/TextToSpeech.android.cs

```diff
@@ -0,0 +1,290 @@
+             await SpeakAsync(text, default(SpeakSettings), cancelToken);
+
+        public static async Task SpeakAsync(string text, SpeakSettings settings, CancellationToken cancelToken = default(CancellationToken))
+        {
+            if (string.IsNullOrEmpty(text))
+            {
+                throw new ArgumentNullException(nameof(text), "Text cannot be null or empty string");
+            }
+
+            Task initialization = null;
+            if (textToSpeech == null)
+            {
+                initialization = Initialize();
+            }
+
+            var parts = SplitText(text, maxSpeechInputLengthDefault);
+
+            await Task.Run(() =>
+            {
+                initialization?.Wait();
+
+                // set the speed and pitch
+                // textToSpeech.SetPitch(settings.Pitch.Value);
+                // textToSpeech.SetSpeechRate(settings.SpeakRate.Value);
+
+                var bundle = Bundle.Empty;
+                string stringutteranceid = null;
+
+                foreach (var t in parts)
+                {
+                    textToSpeech.Speak(t, QueueMode.Flush, bundle, stringutteranceid);
+                }
+            });
+
+            return;
+        }
+
+        private static List<string> SplitText(string text, int maxSpeechInputLengthDefault)
+        {
+            var max = maxSpeechInputLengthDefault;
+
+            if ((int)Build.VERSION.SdkInt >= 18)
+            {
+                max = global::Android.Speech.Tts.TextToSpeech.MaxSpeechInputLength;
+            }
+
+            var count = (text.Length / maxSpeechInputLengthDefault) + 1;
+            Console.WriteLine($"count = {count}");
+            var parts = new List<string>();
+            for (var i = 0; i < count; i++)
+            {
+                var position = i * max;
+                var length = max;
+                if (i == count - 1)
+                {
+                    length = text.Length - position;
+                }
+                Console.WriteLine($"i        = {i}");
+                Console.WriteLine($"position = {position}");
+                Console.WriteLine($"length   = {length}");
+
+                var p = text.Substring(position, length);
+
+                Console.WriteLine($"p = {p}");
+                parts.Add(p);
+            }
+
+            return parts;
+        }
+
+        public static Task<List<Locale>> GetLocalesAsync()
+        {
+            // set up the speech to use the default langauge
+            // if a language is not available, then the default language is used.
+            localedefault = Java.Util.Locale.Default;
+
+            // textToSpeech.SetLanguage(localedefault);
+
+            var locales = new List<Locale>();
+
+            if (textToSpeech != null && TextToSpeech.Initialized)
+            {
+                var sdk = (int)global::Android.OS.Build.VERSION.SdkInt;
+                if (sdk < 21)
```

Comment:
we have platform checks that can be used for this.

Reply:
maybe think about flipping this... if > 21 do this, if == 21, then this.  then you will have a list of locals.